vars/piperInitRunStageConfiguration.groovy

```diff
@@ -57,156 +43,16 @@ void call(Map parameters = [:]) {
         .withMandatoryProperty('stageConfigResource')
         .use()
 
+    // Go logic to check if the step is active
+    String piperGoPath = parameters?.piperGoPath ?: './piper'
+    writeFile(file: ".pipeline/stage_conditions.yaml", text: libraryResource(config.stageConfigResource))
+    piperExecuteBin.checkIfStepActive(script,piperGoPath,".pipeline/stage_conditions.yaml",".pipeline/step_out.json",".pipeline/stage_out.json","_","_")
 
-    config.stages = (readYaml(text: libraryResource(config.stageConfigResource))).stages
-
-    //handling of stage and step activation
-    config.stages.each {stage ->
-
-        String currentStage = stage.getKey()
-        script.commonPipelineEnvironment.configuration.runStep[currentStage] = [:]
-
-        // Always test step conditions in order to fill runStep[currentStage] map
-        boolean anyStepConditionTrue = false
-        stage.getValue().stepConditions?.each {step ->
-            boolean stepActive = false
-            String stepName = step.getKey()
-            step.getValue().each {condition ->
-                Map stepConfig = script.commonPipelineEnvironment.getStepConfiguration(stepName, currentStage)
-                switch(condition.getKey()) {
-                    case 'config':
-                        stepActive = stepActive || checkConfig(condition, stepConfig)
-                        break
-                    case 'configKeys':
-                        stepActive = stepActive || checkConfigKeys(condition, stepConfig)
-                        break
-                    case 'filePatternFromConfig':
-                        stepActive = stepActive || checkForFilesWithPatternFromConfig(script, condition, stepConfig)
-                        break
-                    case 'filePattern':
-                        stepActive = stepActive || checkForFilesWithPattern(script, condition)
-                        break
-                    case 'npmScripts':
-                        stepActive = stepActive || checkForNpmScriptsInPackages(script, condition)
-                        break
-                }
-            }
-            script.commonPipelineEnvironment.configuration.runStep[currentStage][stepName] = stepActive
-
-            anyStepConditionTrue = anyStepConditionTrue || stepActive
-        }
-
-        Map stageConfig = ConfigurationHelper.newInstance(this)
-            .loadStepDefaults([:], currentStage)
-            .mixinStageConfig(script.commonPipelineEnvironment, currentStage)
-            .use()
-
-        boolean runStage
-        if (stageConfig.runInAllBranches == false && (config.productiveBranch != env.BRANCH_NAME)) {
```

Comment:
Not sure what is the logic for stageActive = false here 
Was not able to find stageConfig.runInAllBranches in inner source or open source.